From b7570f668aa9ea8b75ae5f87c646c44a1057747b Mon Sep 17 00:00:00 2001
From: Yassine Oudjana <y.oudjana@protonmail.com>
Date: Sat, 6 Mar 2021 13:33:43 +0400
Subject: [PATCH 10/12] arm64: dts: qcom: pmi8994: Add fuel gauge

Add a node for qcom,pmi8994-fg.

Signed-off-by: Yassine Oudjana <y.oudjana@protonmail.com>
(cherry picked from commit fc793f7e02c121ae59084c60b47448d15045cc9d)
---
 arch/arm64/boot/dts/qcom/pmi8994.dtsi | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/arch/arm64/boot/dts/qcom/pmi8994.dtsi b/arch/arm64/boot/dts/qcom/pmi8994.dtsi
index c7b8ee8ef..972540232 100644
--- a/arch/arm64/boot/dts/qcom/pmi8994.dtsi
+++ b/arch/arm64/boot/dts/qcom/pmi8994.dtsi
@@ -82,6 +82,19 @@ pmi8994_smbchg: charger@1000 {
 			chg_otg: otg-vbus { };
 		};
 
+		pmi8994_fg: fuel-gauge@4000 {
+			compatible = "qcom,pmi8994-fg";
+			reg = <0x4000>;
+
+			interrupts = <0x2 0x40 0x4 IRQ_TYPE_EDGE_RISING>,
+				     <0x2 0x44 0x0 IRQ_TYPE_EDGE_BOTH>;
+			interrupt-names = "soc-delta", "mem-avail";
+
+			power-supplies = <&pmi8994_smbchg>;
+
+			status = "disabled";
+		};
+
 		pmi8994_gpios: gpio@c000 {
 			compatible = "qcom,pmi8994-gpio", "qcom,spmi-gpio";
 			reg = <0xc000>;
-- 
2.56.0

